Added a getting started page (#254)
gsoft-inc/wl-hopper: apps/docs/content/pages/getting-started.mdx

---
title: Getting started
description: Getting started with Workleap Design Tokens
---

> This guide will help you get started with Hopper UI. Hopper UI is a Design System that provides a collection of components, tokens, and icons to help you build applications.

Reviewer: I'm not sure how to word this properly, but i feel like this page is really really too techy, and not consumer oriented. 

There is no such thing as a "Quick Install". 

There are 2 main use case: 
- You're a react dev
- You're a non react dev.

It should be reflected straight away, and maybe redirected to a new page, such as getting-started/react. 

So `/getting-started`

This guide will help you get started with Hopper UI. Hopper UI is a Design System that provides a collection of components, tokens, and icons to help you build applications.

# Installation Guide

Follow one of the guides below to get started with Hopper. 

[ React Application ] [ JavaScript Application ] [ Individual Packages Installation]

# Contribute

Pull requests are welcome. For major changes, please open a [discussion](https://github.com/gsoft-inc/wl-hopper/discussions/new/choose) first to discuss what you would like to change. If you're interested in contributing, check out our [Contributing Guide](https://github.com/gsoft-inc/wl-hopper/blob/main/CONTRIBUTING.md)!
  1 change: 1 addition & 0 deletions1  
[apps/docs/package.json](https://github.com/gsoft-inc/wl-hopper/pull/254/files#diff-d03d2ce97fad5059ad6248fca5f5919994b09bca360f883318c03f1f817ae677)

Then `/getting-started/react`

<Callout>Hopper is compatible with React 18. Not using React, or using an older version than 18? You can still benefit from Hopper, read on to discover how. LINK TO INDIVIDUAL PACKAGE INSTALLATION REACT 16 ICONS</Callout>

La section qui est dans quick install ici

Then `/getting-started/javascript`

la section tokens + svg-icons mergé

Then `/getting-started/individual-packages`

tt les installations

Author: I agree to an extent, I feel that adding "hub" page is an unnecessary click in most cases. How about /getting-started would be the default installation aka react?

Reviewer: sure ya

## Quick Install

To use Hopper in your project, run one of these commands in your terminal.

> <PackageInstallation library="components @hopper-ui/icons @hopper-ui/tokens @hopper-ui/styled-system" mod="prod" />

Reviewer: ![image](https://github.com/gsoft-inc/wl-hopper/assets/38871812/186cfadb-8c57-4994-ad97-5030a960700a)

weird looking in dark mode

> <PackageInstallation library="components @hopper-ui/icons @hopper-ui/tokens @hopper-ui/styled-system" mod="prod" />

Reviewer: For normal react installations, importing @hopper-ui/tokens is not required

> <PackageInstallation library="components @hopper-ui/icons @hopper-ui/tokens @hopper-ui/styled-system" mod="prod" />

Reviewer 2: Packages will be missing `react-aria` `react-aria-components`

After installing Hopper UI packages, you must setup a Provider at the root of your application. This is typically in your _index.tsx_ or _app.tsx_, but can vary based on your specific setup.

> 

Reviewer: manque la section où on doit importer les fichier CSS

> ```jsx

Reviewer 2: I'll give a slightly more specific example for here, as we mention using the Provider at application root.

```
import { HopperProvider } from "@hopper-ui/components";
import { createRoot } from 'react-dom/client';

const root = createRoot(document.getElementById('root'));
root.render(
  < HopperProvider >
    <App />
  </HopperProvider >
);
```

import { HopperProvider } from "@hopper-ui/components";

<HopperProvider colorScheme="light">
    <Button>Hello World</Button>
</HopperProvider>
```

<Callout>Hopper is compatible with React 18. Not using React, or using an older version than 18? You can still benefit from Hopper, read on to discover how.</Callout>

## Detailed Installation

> ### Components

Reviewer 3: I've noticed that there is not much difference in style/size between h3-4. I feel like it ends up being a bit messy or hard to follow. Not sure if this would be something easy to fix or not.

Author: The main issue is that h5 were so small that I bumped them, I'll see what can be done

Author: Here's the current situation: 
h5: 16px
h4: 18px
h3: 20px
h2: 24px
h1: 40px

If h3 goes to 24, h2 would need in theory need to go to 32px its big and close to the h1

![image](https://github.com/gsoft-inc/wl-hopper/assets/361632/fcb6569d-713e-47a1-8603-0f4bee2cd033)

Author: We are using the system fonts, therefore we don't have much control over boldness, as we can't assume a system font will support a semi bold.

> Hopper Components are React Comtponents, uf you are using another framework, or React Native, you should use the [components tokens](/tokens/getting-started/introduction) directly when developping your own components.

Reviewer 3: ```suggestion
Hopper Components are React Components, if you are using another framework, or React Native, you should use the [components tokens](/tokens/getting-started/introduction) directly when developing your own components.
```

#### Installation

> Run one of the following commands in your terminal to install the Hopper Components package.

Reviewer: There is only one command under. I understand you are refering to either npm/pnpm/yarn, but there is 1 command, for 3 different tools. I would phrase it like 

```
Run the following command in your terminal to install the Hopper packages.
```

<PackageInstallation library="components" mod="prod" />

#### Usage

A Provider is required to use Hopper Components. Wrap your application with the HopperProvider component to use our components.

```jsx
import { Button } from "@hopper-ui/components";

<HopperProvider theme="dark">
    <Button>Hello World</Button>
</HopperProvider>
```

### Tokens

Hopper Tokens are a collection of design tokens that should be used to style your application. [Semantic tokens](#) are available for everyone, where [components tokens](#) are designed with non React users in mind.

#### Installation

<PackageInstallation library="tokens" mod="prod" />

#### Usage

##### CSS

Import the tokens in your project by doing the following:

```css
@import "@hopper-ui/tokens/tokens.css";
```

If your application supports a dark mode, import dark mode tokens as well:

```css
@import "@hopper-ui/tokens/dark/tokens.css";
```

You can now use Hopper tokens in your project by using the following syntax:

```css
.my-class {
    background-color: var(--hop-primary-surface-weak);
}
```

##### Fonts

In order to use Hopper fonts in your project, import the following:

```css
@import "@hopper-ui/tokens/fonts.css";
```

> All the necessary font-face declarations are now imported in your project.

Reviewer 3: ```suggestion
All the necessary font-face declarations are now imported into your project.
```

### Icons

#### Installation

<PackageInstallation library="icons" mod="prod" />

#### Usage

You can now use [icons](/icons/react-icons/icon-library) and [rich icons](/icons/react-icons/rich-icon-library) in your project by using the following syntax:

```tsx
import { AddIcon } from "@hopper-ui/icons";
import { ConversationRichIcon } from "@hopper-ui/icons";

<AddIcon />
<ConversationRichIcon />
```

> #### Non-standard installations

Reviewer 2: Non-standard is a bit of a strange name, I don't have much inspiration for a title. Maybe **Javascript installations** or **Framework agnostic installations**.

##### SVG Icons

If your project is not using the React framework, you can use the SVG icons directly. You can find the SVG icons in the `@hopper-ui/svg-icons` package.

<PackageInstallation library="svg-icons" mod="prod" />

You can import the SVG directly into your JavaScript file. Keep in mind that if you plan to use SVG icons as components, you'll need to ensure that your bundler is configured to handle SVG parsing appropriately.

```tsx
import alertIcon from "@hopper-ui/svg-icons/alert-24.svg";
```

Alternatively, you can utilize SVG icons within your CSS in the following manner:

```css
.myComponent {
    background-image: url("@hopper-ui/svg-icons/alert-24.svg");
}
```

> ##### React Icons Standalone Installation

Reviewer 2: I'd remove React Icons because it's redundant 

```suggestion
##### Standalone Installation
```

And I don't quite understand the purpose of this section, isn't it a light version of using Hopper?

Author: @alexasselin008 can you confirm this?

Reviewer: if it can simplify this page we can remove it. But there needs to be orbiter or hopper components installed with the HopperPRovider/ThemePRovider for the icons to work properly

It is recommended to use `@hopper-ui/icons` with `@hopper-ui/components`. The standalone installation procedure is detailed in case you only need the icons, and not the components. This is also the installation process until the components are released.

<PackageInstallation library="icons @hopper-ui/styled-system react-aria-components" mod="prod" />

Import the styles in your project:

```css
@import "@hopper-ui/icons/index.css";
@import "@hopper-ui/styled-system/index.css";
```

> Configuring your application it a little different:

Reviewer 3: ```suggestion
Configuring your application is a little different:
```

```tsx
import { StyledSystemProvider } from "@hopper-ui/styled-system";
import { createRoot } from "react-dom/client";
import App from "./App";

const root = createRoot(document.getElementById("root")!);

root.render(
    <StyledSystemProvider withBodyStyle colorScheme="light">
        <App />
    </StyledSystemProvider>
);
```

Using the icon is the exact same way as with the default package.

```tsx
import { AddIcon } from "@hopper-ui/icons";

<AddIcon />
```

##### React 16 Installation

> If your'e still using React 16, you need to use the `@hopper-ui/icons-react16` package.

Reviewer 3: ```suggestion
If you're still using React 16, you need to use the `@hopper-ui/icons-react16` package.
```

<PackageInstallation library="icons-react16" mod="prod" />

The usage is the same as the default package, you only need to import the icons from a different package.

```tsx

> import { AddIcon } from "@hopper-ui/icons-react16";

Reviewer 2: here, too, the elements can be grouped 
```suggestion
import { AddIcon, ConversationRichIcon } from "@hopper-ui/icons-react16";
```

---
title: Getting started
description: Getting started with Hopper UI.
---

> {/* <Callout>Hopper is compatible with React 18. Not using React, or using an older version than 18? You can still benefit from Hopper, read on to discover how.</Callout> */}

Reviewer 3: is this commented out? how come?

Author: Good call, I thought it was not necessary anymore with the way I restructured things

> This guide will help you getting started with Hopper UI. Hopper UI is a Design System that provides a collection of components, [tokens](/tokens), and [icons](/icons) to help you build applications.

Reviewer 3: ```suggestion
This guide will help you get started with Hopper UI. Hopper UI is a Design System that provides a collection of components, [tokens](/tokens), and [icons](/icons) to help you build applications.
```

Below is the typical installation path, this assumes you are using React. If you have particluar needs you can have a look at these pages:

- [Javascript installation](/pages/getting-started/javascript)
- [Individual package installation](/pages/getting-started/individual-packages)

Run the following command in your terminal to install the Hopper packages.

<PackageInstallation library="components @hopper-ui/icons @hopper-ui/styled-system react-aria react-aria-components" mod="prod" />

> After installing Hopper UI packages, you must setup a Provider at the root of your application. This is typically in your _index.tsx_ or _app.tsx_, but can vary based on your specific setup.

Reviewer: This page would benefit of more headings, similar to orbit

```jsx
import { HopperProvider } from "@hopper-ui/components";
import { createRoot } from 'react-dom/client';

const root = createRoot(document.getElementById('root'));
root.render(

>   < HopperProvider >

Reviewer: Weird spacing it should be 
```
  <HopperProvider>
    <App />
  </HopperProvider>
```

    <App />
  </HopperProvider >
);
```

You will also need to import the Hopper styles in your project. You can do this by importing the following CSS files:

```css

> @import "@hopper-ui/tokens/tokens.css";

Reviewer: c'est pas ça le CSS à importé, c'est 

```
@import "@hopper-ui/components/index.css";
@import "@hopper-ui/icons/index.css";
```
